server/resp/src/main/java/org/infinispan/server/resp/RespRequestHandler.java

```diff
@@ -128,6 +129,10 @@ public void handleChannelDisconnect(ChannelHandlerContext ctx) {
       // Do nothing by default
    }
 
+   public <E> CompletionStage<RespRequestHandler> stageToReturn(CompletionStage<E> stage, ChannelHandlerContext ctx) {
```

Comment:
I question the need of this method... shouldn't all callers know the response type and can use the method with ` BiConsumer<? super E, ByteBufPool> biConsumer` argument as well?

For example MEMORY can use Resp3Response.INTEGER as its third argument.

Reply:
Thinking about it now, it should be a better approach. So even when we implement we need to set the response type.